src/realitycheck/realitycheck.js

```diff
@@ -162,11 +162,15 @@ define(["jquery", "windows/windows", "websockets/binary_websockets", "lodash", '
 
     var oauthLogin = function() {
         logout();
-        if(!local_storage.get("authorize").is_virtual)
-            init().then(function() {
+        if(!local_storage.get("authorize").is_virtual){
+            var p = init();
+            promises.push(p);
```

Comment:
@4p00rv When promises are resolved, are you clearing the array `promises` somewhere? you do not want these promises' resolve to be called later if someone calls init() again when win is already loaded.